packages/sdk/src/chains/EVM/utils/depositFns.ts

```diff
@@ -0,0 +1,296 @@
+  console.log(
+    'Approval before deposit',
+    await getApproved(Number(tokenId), tokenInstance, handlerAddress),
+  );
+
+  // pass data to smartcontract function and create a transaction
+  return executeDeposit(
+    domainId,
+    resourceId,
+    depositData,
+    feeData,
+    bridgeInstance,
+    confirmations,
+    provider,
+    overrides,
+  );
+};
+
+/**
+ * Executes a deposit operation using the specified parameters and returns a contract receipt.
+ *
+ * @example
+ * const domainId = '1';
+ * const resourceId = '0x1234567890123456789012345678901234567890';
+ * const depositData = '0xabcdef0123456789abcdef0123456789abcdef0123456789abcdef0123456789';
+ * const feeData = { type: 'basic', fee: ethers.BigNumber.from('1000000000000000'), feeData: {} }; // fee data is madatory
+ * const bridgeInstance = new Bridge() // Bridge instance from sygma-contracts package;
+ * const confirmations = 3;
+ * const provider = new ethers.providers.JsonRpcProvider();
+ * const overrides = { gasLimit: 200000 };
+ *
+ * executeDeposit(domainId, resourceId, depositData, feeData, bridgeInstance, confirmations, provider, overrides)
+ *   .then((receipt) => console.log('Deposit executed:', receipt))
+ *   .catch((error) => console.error('Error on deposit execution:', error));
+ *
+ * @param {string} domainId - The unique identifier for destination network.
+ * @param {string} resourceId - The resource ID associated with the token.
+ * @param {string} depositData - The deposit data required for the operation.
+ * @param {FeeDataResult} feeData - The fee data result for the deposit operation.
+ * @param {Bridge} bridgeInstance - The bridge instance used to perform the deposit operation.
+ * @param {number} confirmations - The number of confirmations required before the transaction is considered successful.
+ * @param {providers.Provider} provider - The provider used for the Ethereum network connection.
+ * @param {ethers.PayableOverrides} [overrides] - Optional transaction overrides to be applied.
+ * @returns {Promise<ContractReceipt>} A promise that resolves to a contract receipt once the deposit is executed.
+ */
+export const executeDeposit = async (
+  domainId: string,
+  resourceId: string,
+  depositData: string,
+  feeData: FeeDataResult,
+  bridgeInstance: Bridge,
+  confirmations: number,
+  provider: providers.Provider,
+  overrides?: ethers.PayableOverrides,
+): Promise<ContractReceipt> => {
+  try {
+    const gasPrice = await isEIP1559MaxFeePerGas(provider);
+    const gasPriceStringify = typeof gasPrice !== 'boolean' ? gasPrice.toString() : undefined;
+
+    const payableOverrides = {
+      gasPrice: gasPriceStringify,
+      value: feeData.type === 'basic' ? feeData.fee : undefined,
+      ...overrides,
+    };
+    const tx = await bridgeInstance.deposit(
+      domainId,
+      resourceId,
+      depositData,
+      feeData.feeData,
+      payableOverrides,
+    );
+    const depositAction = await tx.wait(confirmations);
```

Comment:
I'm doubtful a SDK should wait for confirmations.

Reply:
It assumes that dev just wanna do deposit and get receipt as a result.
Do you find it more useful to have transaction object instead of transacion receipt?

Reply:
ok, now the fuction are returning `ContractTransaction` instead of `ContractReceipt`